sdk/test/resource/resource_test.cc

```diff
@@ -61,17 +61,17 @@ TEST(ResourceTest, create_without_servicename)
 
 TEST(ResourceTest, create_with_servicename)
 {
-  opentelemetry::sdk::resource::ResourceAttributes expected_attributes = {
+  ResourceAttributes expected_attributes = {
       {"version", (uint32_t)1},
       {"cost", 234.23},
-      {"telemetry.sdk.language", "cpp"},
-      {"telemetry.sdk.name", "opentelemetry"},
-      {"telemetry.sdk.version", OPENTELEMETRY_SDK_VERSION},
-      {"service.name", "backend"},
+      {SemanticConventions::GetAttributeTelemetrySdkLanguage(), "cpp"},
```

Comment:
Actually to elaborate on my idea - you probably do not even need to change anything. Keep passing original string literals. We have an assignment operator in the `ResourceAttributes` implementation, that could have done the semantic transformation depending on what exporter / system is being used. Then you'd just completely avoid calling getters. You'd pre-configure your `ResourceAttributes` by specifying destination convention (like, imagine - your `ResourceAttributes` class can translate from English to French, and you pass in English keys, but they get magically translated to French, because your system "locale" is French).

If you need concrete example, I can send some WIP code to illustrate the model?

Reply:
Thanks @maxgolov for the elaboration. I can try implementing this model, but if you can provide a more concrete implementation example it would be helpful.

Reply:
We can definitely avoid getters. OR - at least, it all boils down to a single method that allows to transform from schema-agnostic common `attribute name` or `attribute id` to semantic- / schema- specific attribute name, the way it shows up on wire. I feel like more concrete code example (based off above constexpr hash calculation) would be helpful?

Reply:
@maxgolov  I have done the changes (hopefully) as per your suggestion :)
Have kept both the alternatives for now so that we can compare both and finalize.
`semantic_conventions.h` -> the suggested change.
`semantic_conventions_old.h` -> Initial implementation